test/easy-sql-test.js

```diff
@@ -2,13 +2,742 @@
+            connect: function() {},
+            request: function() {
+              return {
+                input: function() {},
+                execute: executeFunctionSpy
+              };
+            }
+          };
+        }
+      };
+
+      var EasySqlTest = proxyquire('../dist/easy-sql-test.js', {
+        mssql: sqlStub
+      });
+
+      var easySqlProxy = new EasySqlTest(emptyDbConfig),
+          callbackFunction = 'callbackFunction',
+          storProcName = 'storProcName';
+
+      easySqlProxy.connectionOpen();
+      easySqlProxy._executeStorProc(storProcName, {}, callbackFunction);
+      easySqlProxy._executeStorProc(storProcName);
+
+      var firstCall = executeFunctionSpy.getCall(0),
+          secondCall = executeFunctionSpy.getCall(1);
+
+      expect(firstCall.calledWithExactly(storProcName, callbackFunction))
+        .to.be.true;
+
+      expect(secondCall.calledWithExactly(storProcName, undefined))
+         .to.be.true;
+    });
+  });
+
+  describe('_query() tests', function(){
+    it('request.query called correctly', function() {
+      var expectedCallback = function() { return 'a' },
+          queryFunctionSpy = sinon.spy(function() {});
+          expectedArgs = {
+            a: 'a',
+            b: 'b'
+          };
+
+      var sqlStub = {
+        Connection: function() {
+          return {
+            connect: function() {},
+            request: function() {
+              return {
+                query: queryFunctionSpy
+              };
+            }
+          };
+        }
+      };
+
+      var EasySqlTest = proxyquire('../dist/easy-sql-test.js', {
+        mssql: sqlStub
+      });
+
+      var easySqlProxy = new EasySqlTest(emptyDbConfig);
+      easySqlProxy.connectionOpen();
+
+      easySqlProxy._query(expectedArgs, expectedCallback);
+      easySqlProxy._query(undefined, expectedCallback);
+      easySqlProxy._query(expectedArgs, undefined);
+      easySqlProxy._query();
+
+      var firstCall = queryFunctionSpy.getCall(0),
+          secondCall = queryFunctionSpy.getCall(1),
+          thirdCall = queryFunctionSpy.getCall(2),
+          fourthCall = queryFunctionSpy.getCall(3);
+
+      expect(queryFunctionSpy.called).to.be.true;
+      expect(firstCall.calledWithExactly(expectedArgs, expectedCallback))
+        .to.be.true;
+      expect(secondCall.calledWithExactly(undefined, expectedCallback))
+        .to.be.true;
+      expect(thirdCall.calledWithExactly(expectedArgs, undefined)).to.be.true;
+      expect(fourthCall.calledWithExactly(undefined, undefined)).to.be.true;
+    });
+  });
+
+  describe('_convertQueriesToTestSteps() tests', function() {
+    var easySqlTest,
+        prepQuery;
+
+    function assertionCallback(error) {
+      if (error) {
+        this._errorCallback(error);
+      }
+    };
+
+    beforeEach(function() {
+      easySqlTest = new EasySqlTest(emptyDbConfig);
+      prepQuery = undefined;
+    });
+
+    it('with no prepQuery', function() {
```

Comment:
if we change prepQuery to queries then we need to change name here. Also tests should reflect decisions I mentioned ^ in case if we want to avoid skipping queries which are empty or "malformed"